### PR TITLE
fix(ButtonGroup): Add `width: 100%` for overflow handling

### PR DESCRIPTION
**Pull request by mayank99**
This will address https://github.com/iTwin/iTwinUI-react/issues/602

### CODE REVIEW

**Review by FlyersPh9 — APPROVED** [review]

File: src/button/button-group.scss [hunk 1]
```diff
@@ -106,3 +106,8 @@
     }
   }
 }
+
+@mixin iui-button-group-overflow-x {
```
Comment (veekeys) [inline src/button/button-group.scss:110]: so when would you apply that class in react?

File: src/button/button-group.scss [hunk 2]
```diff
@@ -106,3 +106,8 @@
...
+  max-width: revert;
```
Comment (veekeys) [inline src/button/button-group.scss:111]: Remind again, why max-width is always needed?

File: src/button/button-group.scss (hunk 1 above)
Comment (mayank99) [reply src/button/button-group.scss:110]: `{ 'iui-button-group-overflow-x': !!overflowButton && orientation === 'horizontal' }`

File: src/button/button-group.scss (hunk 2 above)
Comment (mayank99) [reply src/button/button-group.scss:111]: Hmm I think it could be removed now actually

File: src/button/button-group.scss (hunk 1 above)
Comment (veekeys) [reply src/button/button-group.scss:108]: and if you always keep it applied, whats breaking?

File: src/button/button-group.scss [hunk 3]
```diff
@@ -10,9 +10,7 @@
   align-items: center;
   isolation: isolate;
 
-  @if $orientation == 'horizontal' {
-    max-width: 100%;
-  } @else {
+  @if $orientation != 'horizontal' {
```
Comment (veekeys) [inline src/button/button-group.scss:13]: so just `== 'vertical'`

Comment (mayank99) [reply src/button/button-group.scss:13]: This is the personal convention I chose for this component. See https://github.com/iTwin/iTwinUI/pull/503#discussion_r816922553

Comment (veekeys) [reply src/button/button-group.scss:13]: huh? But dont you want to apply that rule only when vertical is chosen (and not some random stuff) ?

File: src/button/button-group.scss (hunk 1 above)
Comment (veekeys) [reply src/button/button-group.scss:108]: But if I remember correctly, wehad 100% before your previous changes...no?

File: src/button/button-group.scss (hunk 3 above)
Comment (mayank99) [reply src/button/button-group.scss:13]: by that logic you would never use `@else`

Comment (veekeys) [reply src/button/button-group.scss:13]: I really dont understand , where are you going with this or not having else :DD
But w/e, its minor not worth spending hours..

**Pull request merged**